src/main.cpp

```diff
@@ -0,0 +1,49 @@
+/****************************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Martin Grimme <martin.grimme@gmail.com>
+** All rights reserved.
+**
+** This file is part of the btrfs-balancer package.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+
+#include "dbusconnector.h"
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <backgroundactivity.h>
+
+int main(int argc, char *argv[])
+{
+    QCoreApplication app(argc, argv);
+    DBusConnector dbusConnector;
+
```

Comment:
Right now, all the warnings will be logged to the systemd journal (hopefully). Most of the messages sound like they are being printed to the user on the console ("please plug in charger", etc..). Would it make sense to have a mode (e.g. via command line flag) where the tool behaves like the old script? This way, you can run it interactively on the console (in this case, it wouldn't register the D-Bus service, but just call the balancing operation directly, and output all messages to the console). This might be useful for debugging and during development.